zipline/pipeline/loaders/buyback_auth.py

```diff
@@ -0,0 +1,103 @@
+"""
+Reference implementation for EarningsCalendar loaders.
+"""
+
+from ..data.buyback_auth import (
+    CashBuybackAuthorizations,
+    ShareBuybackAuthorizations
+)
+from .events import EventsLoader
+from zipline.utils.memoize import lazyval
+
+
+BUYBACK_ANNOUNCEMENT_FIELD_NAME = 'buyback_date'
+SHARE_COUNT_FIELD_NAME = 'share_count'
+CASH_FIELD_NAME = 'cash'
+
+
+class CashBuybackAuthorizationsLoader(EventsLoader):
+    """
+    Reference loader for
+    :class:`zipline.pipeline.data.earnings.CashBuybackAuthorizations`.
+
+    events_by_sid: dict[sid -> pd.DataFrame(knowledge date,
+    event date, cash value)]
+
+    """
+
+    def __init__(self,
+                 all_dates,
+                 events_by_sid,
+                 infer_timestamps=False,
+                 dataset=CashBuybackAuthorizations):
+        super(CashBuybackAuthorizationsLoader, self).__init__(
+            all_dates,
+            events_by_sid,
+            infer_timestamps=infer_timestamps,
+            dataset=dataset,
+        )
+
+    @property
+    def expected_cols(self):
```

Comment:
`abstractpropery` and `abstractmethod` only check that the value is in the class dict at all. You should make the abc have an abstract property to get the checks, but the subclasses can just put a frozen set in the class dict to satisfy the check.